agent/azure-monitor-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/LogDataMapper.java

```diff
@@ -19,114 +19,57 @@
+  private static final Logger logger = LoggerFactory.getLogger(LogDataMapper.class);
 
   private static final AttributeKey<String> AI_OPERATION_NAME_KEY =
       AttributeKey.stringKey("applicationinsights.internal.operation_name");
 
-  private static final OperationLogger exportingLogLogger =
-      new OperationLogger(LoggerExporter.class, "Exporting log");
-
-  private final TelemetryClient telemetryClient;
   private final boolean captureLoggingLevelAsCustomDimension;
+  private final Consumer<AbstractTelemetryBuilder> defaultsPopulator;
 
-  // TODO (trask) could implement this in a filtering LogExporter instead
-  private volatile Severity threshold;
-
-  public LoggerExporter(
-      TelemetryClient telemetryClient,
-      Severity threshold,
-      boolean captureLoggingLevelAsCustomDimension) {
-    this.telemetryClient = telemetryClient;
-    this.threshold = threshold;
+  public LogDataMapper(
+      boolean captureLoggingLevelAsCustomDimension,
+      Consumer<AbstractTelemetryBuilder> defaultsPopulator) {
     this.captureLoggingLevelAsCustomDimension = captureLoggingLevelAsCustomDimension;
+    this.defaultsPopulator = defaultsPopulator;
   }
 
-  public void setThreshold(Severity threshold) {
-    this.threshold = threshold;
-  }
-
-  @Override
-  public CompletableResultCode export(Collection<LogData> logs) {
-    if (CoreUtils.isNullOrEmpty(TelemetryClient.getActive().getInstrumentationKey())) {
-      logger.debug("Instrumentation key is null or empty. Fail to export logs.");
-      return CompletableResultCode.ofFailure();
-    }
-
-    boolean failure = false;
-    for (LogData log : logs) {
-      logger.debug("exporting log: {}", log);
-      try {
-        internalExport(log);
-        exportingLogLogger.recordSuccess();
-      } catch (Throwable t) {
-        exportingLogLogger.recordFailure(t.getMessage(), t);
-        failure = true;
-      }
-    }
-
-    // batching, retry, throttling, and writing to disk on failure occur downstream
-    // for simplicity not reporting back success/failure from this layer
-    // only that it was successfully delivered to the next layer
-    return failure ? CompletableResultCode.ofFailure() : CompletableResultCode.ofSuccess();
-  }
-
-  @Override
-  public CompletableResultCode flush() {
-    return CompletableResultCode.ofSuccess();
-  }
-
-  @Override
-  public CompletableResultCode shutdown() {
-    return CompletableResultCode.ofSuccess();
-  }
-
-  private void internalExport(LogData log) {
-    int severity = log.getSeverity().getSeverityNumber();
-    int threshold = this.threshold.getSeverityNumber();
-    if (severity < threshold) {
-      return;
-    }
-
+  public void map(LogData log, Consumer<TelemetryItem> consumer) {
     String stack = log.getAttributes().get(SemanticAttributes.EXCEPTION_STACKTRACE);
     if (stack == null) {
-      trackMessage(log);
+      consumer.accept(trackMessage(log));
```

Comment:
i noticed all the track method return TelemetryBuilder.build() now, i.e. TelemetryItem, used to be void and then consumer will accept it.. i'm not sure I understand the benefit of doing this way.

Reply:
please rename all the track methods to something related to mapping since they no longer do the actual sending.

Reply:
done